src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1,79 +1,1419 @@
-    supersetService(SUPERSET_PLANS_TABLE_SLICE, supersetParams).then(
-      (result: PlanRecordResponse[]) => fetchPlansActionCreator(result)
-    );
+  public async componentDidMount() {
+    const {
+      allJurisdictionIds,
+      fetchAllJurisdictionIdsActionCreator,
+      fetchJurisdictionsActionCreator,
+      fetchPlansActionCreator,
+      isDraftPlan,
+      isFinalizedPlan,
+      loadedJurisdictionIds,
+      planId,
+      planById,
+      supersetService,
+    } = this.props;
+
+    // GET LIST OF ALL JURISDICTIONS
+    let allJurIds: string[] = [];
+    if (!allJurisdictionIds.length || allJurisdictionIds.length === loadedJurisdictionIds.length) {
+      // todo - is there a better way to fetch a list of ALL jurisdiction ids?
+      await supersetFetch(SUPERSET_JURISDICTIONS_DATA_SLICE, { row_limit: 3000 }).then(result => {
+        // populate array of unique `id`s
+        if (result && result.length) {
+          for (const j of result) {
+            if (allJurIds.indexOf(j.id) === -1) {
+              allJurIds.push(j.id);
+            }
+          }
+        }
+        return fetchAllJurisdictionIdsActionCreator([...allJurIds]);
+      });
+    } else {
+      allJurIds = [...allJurisdictionIds];
+    }
+
+    // GET PLAN
+    if (planId && !planById) {
+      await OpenSrpPlanService.read(planId)
+        .then((plan: PlanPayload) => {
+          const planRecord = extractPlanRecordResponseFromPlanPayload(plan);
+          if (planRecord) {
+            return fetchPlansActionCreator([planRecord]);
+          }
+        })
+        .catch(err => err);
+    } else if (isDraftPlan && planById) {
+      this.setState({ newPlan: planById });
+    }
+
+    // GET PLAN JURISDICTIONS associated with this plan
+    let planJurisdictionIdsToGet: string[] = [];
+    let otherJurisdictionIdsToGet: string[] = [];
+    if (planId && isFinalizedPlan) {
+      const pivotParams = superset.getFormData(500, [
+        { comparator: planId, operator: '==', subject: 'plan_id' },
+      ]);
+      await supersetService(SUPERSET_PLAN_STRUCTURE_PIVOT_SLICE, pivotParams).then(
+        relevantJuridictions => {
+          if (!relevantJuridictions) {
+            return new Promise((resolve, reject) => reject());
+          }
+
+          // define jurisdictions not in the store
+          planJurisdictionIdsToGet = relevantJuridictions
+            .map((j: any) =>
+              !loadedJurisdictionIds.includes(j.jurisdiction_id) ? j.jurisdiction_id : null
+            )
+            .filter((j: string | null) => j && strict.length);
+
+          if (!planJurisdictionIdsToGet.length) {
+            return new Promise(resolve => resolve());
+          }
+
+          // build superset adhoc filter expression
+          let sqlFilterExpression = '';
+          for (let i = 0; i < planJurisdictionIdsToGet.length; i += 1) {
+            const jurId = planJurisdictionIdsToGet[i];
+            if (i) {
+              sqlFilterExpression += ' OR ';
```

Comment:
I think in many cases using `IN` vs `OR` is usually faster/more efficient.  Do you think we should do that here?  Source: https://stackoverflow.com/a/3074731

Reply:
Oooh, sure thing

Reply:
@moshthepitt If i could get an example of the `IN` syntax being used in SupersetConnector's `sqlExpression` option that would be super helpful.